snuba/datasets/table_storage.py

```diff
@@ -10,11 +9,10 @@
 from snuba.processor import MessageProcessor
 from snuba.snapshots.loaders import BulkLoader
 from snuba.writer import BatchWriter
+from snuba.utils.streams.types import Topic
 
 
-@dataclass(frozen=True)
-class KafkaTopicSpec:
-    topic_name: str
+class KafkaTopic(Topic):
```

Comment:
> Why not moving this to streams/types as well ?

No real reason, I think it makes sense to move it.

> Also I think you might have to flag this as a dataclass as well. I ran into problems using dataclass.replace over a class that extended a dataclass. In that case I had overridden the constructor, so in this case it may work. Worth checking.

Will do, thanks for pointing that out.

Reply:
Going to remove this per https://github.com/getsentry/snuba/pull/686/files#r366650502, but for posterity: the subclass has to also include the decorator to pick up extended arguments, they are not implicitly added via any metaclass magic:

```python
>>> KafkaTopic('asdf')
KafkaTopic(name='asdf')
>>> KafkaTopic('asdf', replication_factor=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'replication_factor'
```